gateway/1.5/modules/ROOT/pages/policies-pdk-configure-features.adoc

```diff
@@ -1,5 +1,589 @@
+
+== Use DataWeave Expressions
+
+If you define a `string` type configuration parameter with the `dataweave` format in your xref:policies-pdk-create-schema-definition.adoc[], the parameter maps to a Rust `Expression` type parameter.
+ 
+To solve the expression, await the header state and pass it as a parameter to the expression. After the expression is solved, and the potential errors are handled, you can obtain the result value. Because the value can be of different types, cast the value using the provided functions:
+
+[source,Rust]
+----
+fn is_null(&self) -> bool;
+fn as_bool(&self) -> Option<bool>;
+fn as_f64(&self) -> Option<f64>;
+fn as_str(&self) -> Option<&str>;
+fn as_slice(&self) -> Option<&[Value]>;
+fn as_object(&self) -> Option<&Object>;
+----
+
+The functions return an `Option` containing the value if it was properly casted or `None` if the value didn't belong to the casted type. For example, the following code snippet, tests what type the value resolves as:
+
+[source,Rust]
+----
+async fn response_filter(response_state: ResponseState, config: &Config) {
+    let headers_state = response_state.into_headers_state().await;
+    
+    match config.dw_property.resolve_on_headers(&headers_state) {
+        Ok(value) => {
+            if value.is_null() {
+                logger::info!("Datewave expression resolved to null!");
+            }
+            else if let Some(val) = value.as_bool() {
+                logger::info!("Datewave expression resolved to boolean: {val}!");
+            }
+            else if let Some(val) = value.as_f64() {
+                logger::info!("Datewave expression resolved to f64: {val}!");
+            }
+            else if let Some(val) = value.as_str() {
+                logger::info!("Datewave expression resolved to str: {val}!");
+            }
+            else if let Some(val) = value.as_slice() {
+                logger::info!("Datewave expression resolved to slice: {val:?}!");
+            }
+            else if let Some(val) = value.as_object() {
+                logger::info!("Datewave expression resolved to object: {val:?}!");
+            }
+        },
+        Err(err) => {
+            logger::warn!("Failed to resolve the dataweave expression: {err}!")
+        }
+    }
+}
+----
+
+[[stop-request-execution]]
+== Stop Request Execution
+
+During request processing, you can abort the request and send a response.
+
+Do this by returning a `Response` object from your `on_request` wrapped functions:
+
+[source,Rust]
+----
+async fn request_filter(_request_state: RequestState) -> Response {
+    Response::new(401)
+        .with_headers(vec![("WWW-Authenticate".to_string(), "Bearer realm=\"oauth2\"".to_string())])
+        .with_body(r#"{ "error": "token was not present"}"#)
+}
+----
+
+This function is not very useful as it intercepts the requests no matter what. You can block or allow requests to reach the upstream service using a `Flow`. A `Flow` in an `enum` value with two possible values:
```

Comment:
If this function is not useful, we should present it as an alternative and show the second option (You can block or allow requests to reach the upstream service using a `Flow`) as the first one. 

Also, I'd rephrase the introduction to this function to something like: "In case you need to intercept and stop the request under any circumstances, return a `Response` object from your `on_request` wrapped functions"